firebase-sessions/src/test/kotlin/com/google/firebase/sessions/api/CrashEventReceiverTest.kt

```diff
@@ -0,0 +1,122 @@
+import com.google.firebase.sessions.testing.FakeEventGDTLogger
+import com.google.firebase.sessions.testing.FakeFirebaseApp
+import com.google.firebase.sessions.testing.FakeFirebaseInstallations
+import com.google.firebase.sessions.testing.FakeProcessDataManager
+import com.google.firebase.sessions.testing.FakeSettingsProvider
+import com.google.firebase.sessions.testing.FakeTimeProvider
+import com.google.firebase.sessions.testing.FakeUuidGenerator
+import kotlin.time.Duration.Companion.minutes
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.asCoroutineDispatcher
+import kotlinx.coroutines.test.runCurrent
+import kotlinx.coroutines.test.runTest
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@OptIn(ExperimentalCoroutinesApi::class)
+@RunWith(AndroidJUnit4::class)
+class CrashEventReceiverTest {
+  @Test
+  fun notifyCrashOccurredOnForegroundOnly() = runTest {
+    // Setup
+    val fakeFirebaseApp = FakeFirebaseApp()
+    val fakeEventGDTLogger = FakeEventGDTLogger()
+    val firebaseInstallations = FakeFirebaseInstallations("FaKeFiD", "FakeAuthToken")
+    val fakeTimeProvider = FakeTimeProvider()
+    val sessionGenerator = SessionGenerator(fakeTimeProvider, FakeUuidGenerator())
+    val localSettingsProvider = FakeSettingsProvider(true, null, 100.0)
+    val remoteSettingsProvider = FakeSettingsProvider(true, null, 100.0)
+    val sessionsSettings = SessionsSettings(localSettingsProvider, remoteSettingsProvider)
+    val publisher =
+      SessionFirelogPublisherImpl(
+        fakeFirebaseApp.firebaseApp,
+        firebaseInstallations,
+        sessionsSettings,
+        eventGDTLogger = fakeEventGDTLogger,
+        TestOnlyExecutors.background().asCoroutineDispatcher() + coroutineContext,
+      )
+    val fakeDataStore =
+      FakeDataStore(
+        SessionData(
+          SessionDetails(SESSION_ID_INIT, SESSION_ID_INIT, 0, fakeTimeProvider.currentTime().ms),
+          fakeTimeProvider.currentTime(),
+        )
+      )
+    val sharedSessionRepository =
+      SharedSessionRepositoryImpl(
+        sessionsSettings = sessionsSettings,
+        sessionGenerator = sessionGenerator,
+        sessionFirelogPublisher = publisher,
+        timeProvider = fakeTimeProvider,
+        sessionDataStore = fakeDataStore,
+        processDataManager = FakeProcessDataManager(),
+        backgroundDispatcher =
+          TestOnlyExecutors.background().asCoroutineDispatcher() + coroutineContext,
+      )
+    CrashEventReceiver.sharedSessionRepository = sharedSessionRepository
+
+    runCurrent()
+
+    // Process starts in the background
+    assertThat(sharedSessionRepository.isInForeground).isFalse()
+
+    // This will not update background time since the process is already in the background
+    CrashEventReceiver.notifyCrashOccurred()
+    assertThat(sharedSessionRepository.localSessionData.backgroundTime)
+      .isEqualTo(fakeTimeProvider.currentTime())
```

Comment:
IIUC this assertThat is correct because the value in `fakeTimeProvider` isn't updated before `notifyCrashOccurred`.

I think a more accurate test would capture the original value, update it before `notifyCrashOccurred` and then verify equality with the value captured earlier.